connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -407,10 +407,12 @@ private void resumeAll() {
         for (TopicPartition tp : consumer.assignment())
             if (!context.pausedPartitions().contains(tp))
                 consumer.resume(singleton(tp));
+        onResume();
     }
 
     private void pauseAll() {
         consumer.pause(consumer.assignment());
+        onPause();
```

Comment:
That's a good point. This is somewhat straightforward to fix:
*  in `pauseAll()`, between pausing the consumer and the status callback, `closePartitions()` (which will take care of the commit also).
* in `resumeAll()`, as the very first thing call `openPartitions(consumer.assignment())`.

but there is interplay with the context and I'm not sure what the behavior should be. There may be some partitions that are currently paused by the sink task. The task can also pause/resume partitions using the context while we are paused. While paused, `context.assignment()` will continue to return the assigned partitions, despite us having called `task.close(partitions)` for them.

I think it's probably best to just pause record delivery to the task here. I will add an offset commit though.